src/cccid.rs

```diff
@@ -48,16 +48,27 @@ const CCC_TMPL: &[u8] = &[
     0x00, 0xfe, 0x00,
 ];
 
-/// Cardholder Capability Container (CCC) Identifier
+/// Cardholder Capability Container (CCC) Identifier card ID
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
-pub struct CCCID(pub [u8; YKPIV_CCCID_SIZE]);
+pub struct CccCardId(pub [u8; YKPIV_CCCID_SIZE]);
+
+/// Cardholder Capability Container (CCC) Identifier
+#[derive(Copy, Clone)]
+pub struct CCC(pub [u8; YKPIV_CCC_SIZE]);
+
+impl CCC {
+    /// Return CardId component of CHUID
+    pub fn cccid(&self) -> Result<CccCardId, Error> {
+        let mut cccid = [0u8; YKPIV_CCCID_SIZE];
+        cccid.copy_from_slice(&self.0[CCC_ID_OFFS..(CCC_ID_OFFS + YKPIV_CCCID_SIZE)]);
+        Ok(CccCardId(cccid))
+    }
 
-impl CCCID {
     /// Generate a random CCCID
-    pub fn generate() -> Result<Self, Error> {
+    pub fn generate() -> Result<CccCardId, Error> {
```

Comment:
It seems weird for this method to not return `Self`

Reply:
It's no longer the focus of the struct. It's like the fascn getter on the CHUID class.  I could see ditching CccCardId and returning an array